src/OpenRiaServices.Hosting.AspNetCore/Test/OpenRiaServices.Hosting.AspNetCore.Test/OperationInvokerTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Cities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenRiaServices.Hosting.AspNetCore.Operations;
+using OpenRiaServices.Server;
+using OpenRiaServices.Server.Test;
+
+namespace OpenRiaServices.Hosting.AspNetCore
+{
+    [TestClass]
+    public class OperationInvokerTests
+    {
+        [TestMethod]
+        public async Task TestCancelInvokeOperationInvoker()
+        {
+            var operation = GetOperationEntry();
+            var serializationHelper = GetSerializationHelper();
+            var operationInvoker = new InvokeOperationInvoker(operation, serializationHelper);
+            await TestCancelOperationInvoker(operationInvoker);
+        }
+
+        [TestMethod]
+        public async Task TestCancelQueryOperationInvoker()
+        {
+            var operation = GetOperationEntry();
+            var serializationHelper = GetSerializationHelper();
+            var operationInvoker = new QueryOperationInvoker<City>(operation, serializationHelper);
+            await TestCancelOperationInvoker(operationInvoker);
+        }
+
+        private static async Task TestCancelOperationInvoker(OperationInvoker operationInvoker)
+        {
+            var context = GetHttpContext();
+            var cts = new CancellationTokenSource();
+            context.RequestAborted = cts.Token;
+            var context2 = GetHttpContext();
+            var cts2 = new CancellationTokenSource();
+            context2.RequestAborted = cts2.Token;
+
+            cts2.Cancel();
+            await operationInvoker.Invoke(context);
+            await operationInvoker.Invoke(context2);
+            Assert.IsTrue(context.Response.ContentLength is not null, "A response should have been written");
```

Comment:
It is not directly clear what 2 cases you are testing.

Maybe you can have separte methods  or code blocks.

I think relevant cases are:
1. CancellationToken is cancelled when calling invoker
2. CancellationToken is not cancelled when calling invoker but is cancelled later when DomainService code is called 
     2.a The domainservice's Invoke/Submit/Query[Async] cancels token and throws exception
     2.b The domainservice's Invoke/Submit/Query[Async] cancels token and returns with a successfull result


I think all tests will be easier if you create a custom DomainService for these specific tests with parameters to control the behaviour (throw or not, etc)